_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -369,14 +508,9 @@ mfxStatus VideoDECODEVP9_HW::Close()
 
 void VideoDECODEVP9_HW::ResetFrameInfo()
 {
-    for (mfxU8 i = 0; i < sizeof(m_frameInfo.ref_frame_map)/sizeof(m_frameInfo.ref_frame_map[0]); i++)
-    {
-        const UMC::FrameMemID oldMid = m_frameInfo.ref_frame_map[i];
-        if (oldMid >= 0)
-        {
-            m_FrameAllocator->DecreaseReference(oldMid);
-        }
-    }
+    if (m_framesStorage)
```

Comment:
Useless if. You should add the destructor to the class. And you missed the cleanup in the destructor. CompleteFrames removes decoded frame from the map and frees the resources. IsAllFramesCompleted should check that the map is empty. The destructor should clean up the map.